ZFSin/zfs/module/zfs/zfs_ioctl.c

```diff
@@ -5909,6 +5909,18 @@ zfs_ioctl_register_dataset_modify(zfs_ioc_t ioc, zfs_ioc_legacy_func_t *func,
 							  DATASET_NAME, B_TRUE, POOL_CHECK_SUSPENDED | POOL_CHECK_READONLY);
 }
 
+int
+zfs_ioc_unregister_fs(void) {
+	//IoUnregisterFsRegistrationChange(WIN_DriverObject, DriverNotificationRoutine);
+	dprintf("%s\n", __func__);
+	IoUnregisterFileSystem(fsDiskDeviceObject);
+	IoDeleteDevice(fsDiskDeviceObject);
+	IoDeleteDevice(ioctlDeviceObject);
```

Comment:
Don't we create 3 devices? disk, fs and ioctl? Should we have the 3rd here for completeness?

Reply:
I think the diskDeviceObject what you are referencing is created per dataset in `zfs_windows_mount`, isn't it? In `zfs_attach` the `fsDiskDeviceObject` and `ioctlDeviceObject` are the only devices that are created (If I haven't overlooked anything again :))